src/utility_function_parameters.jl

```diff
@@ -0,0 +1,274 @@
+                                 input_uk::Matrix{T},input_eu::Matrix{T},input_world::Matrix{T}) where {T <: Real}
+
+    parameters = Array{Float64}(undef, length(log_price_uk), length(log_price_uk), 3)
+
+    for i in axes(input_uk, 1)
+        for j in axes(input_uk, 2)
+            parameters[i,j,:] .= parameters_by_region(elasticity, log_price_uk[j], log_price_eu[j], log_price_world[j],
+                                                input_uk[i,j], input_eu[i,j], input_world[i,j])
+        end
+    end
+
+    return parameters
+
+end
+
+"""
+  Compute the total parameter (γM) for the firms input utility function.
+
+  Matlab code reference ComputeTheta.m line 251
+"""
+function total_input_parameters(log_price_index::Vector{T}, input::Vector{T},
+                            capital::T, demand0::T, output::T, labor::T, log_wages::T, elasticity::T, tau::T) where T
+
+    length(log_price_index) == length(input) || error()
+    tauP = tauPdMu(elasticity, log_price_index, input, capital, demand0, output, labor, log_wages, tau)
+
+    parameters = Vector{T}(undef, length(log_price_index))
+
+    for i in axes(log_price_index,1)
+        parameters[i] = weight_kernel(input[i], exp(log_price_index[i])/tauP, elasticity)
+    end
+
+    return parameters
+
+end
+
+"""
+  Compute the total parameter (γH) for the firms labor utility function.
+
+  Matlab code reference ComputeTheta.m line 249
+"""
+function total_labor_parameters(log_price_index::Vector{T}, input::Vector{T},
+                            capital::T, demand0::T, output::T, labor::T, log_wages::T, elasticity::T, tau::T) where T
+
+    length(log_price_index) == length(input) || error()
+    tauP = tauPdMu(elasticity, log_price_index, input, capital, demand0, output, labor, log_wages, tau)
+    return weight_kernel(labor, exp(log_wages) / tauP, elasticity)
+
+end
+
+"""
+  Compute the total parameter (γK) for the firms capital utility function.
+
+  Matlab code reference ComputeTheta.m line 254
+"""
+function total_capital_parameters(log_price_index::Vector{T}, input::Vector{T},
+                            capital::T, demand0::T, output::T, labor::T, log_wages::T, elasticity::T, tau::T) where T
+
+    length(log_price_index) == length(input) || error()
+    tauP = tauPdMu(elasticity, log_price_index, input, capital, demand0, output, labor, log_wages, tau)
+    tauY = (1 - tau) * output / demand0
+    return capital ^ elasticity * (tauY / tauP) ^ (elasticity - 1.0)
+
+end
+
+function weight_kernel(a::T, b::T, elasticity::T) where {T <: Real}
+
+    return a / b ^ (1.0 - elasticity)
+
+end
+
+function log_weight_kernel(a::T, b::T, elasticity::T) where {T <: Real}
+
+    return log(a) + (elasticity - 1.0) * log(b)
+
+end
+
+function sum_kernel(var::Vector{T}, logP::Vector{T}, elasticity::T) where {T <: Real}
+
+    s = 0.0
+    for i in axes(logP,1)
+        s += var[i] ^ (1.0 / elasticity) * exp((elasticity - 1.0) * logP[i] / elasticity)
+    end
+
+    return s
```

Comment:
Alternatively, the entire reduction can be done in one line:
```suggestion
    return sum(x -> x[1] ^ (1 / elasticity) * exp((elasticity - 1) * x[2] / elasticity), zip(var, logP))
```